### ISSUE TITLE
Multi site Support

### ISSUE THREAD

**Issue by suresh-kumar-ogilvy** [chapter-three/next-drupal#332]
Hi Good day 

Does this package support multisite drupal installation(separate database)? 
In other words, every domain has its own .env file so we can load the different content by site same source code.

Thanks

**Comment by shadcn** [comment]
Yes. You can use one Drupal back-end and have multiple Next.js front-ends. 

All the examples in the /examples dir are actually built from one Drupal site.

**Comment by suresh-kumar-ogilvy** [comment]
Thanks, @shadcn I Understand. We are thinking of rolling out around 20 sub-sites from one central site. is there a way to load the env config per domain/host? then don't need to set up 20 front-end servers.

**Comment by shadcn** [comment]
do you mean for the `DrupalClient`?

You should be able to do something like this:


```ts
const baseUrl = doSomethingToRetrieveBaseUrl()
export const drupal = new DrupalClient(baseUrl)
```

**Comment by suresh-kumar-ogilvy** [comment]
yes something like this , based on host

**Comment by shadcn** [comment]
Yeah you should be able to do this.

Another solution you could explore is having a Next.js site that acts as a server between your clients and Drupal. 

All your clients talk to this proxy and this proxy decides what config/data to send based on the host.

![Untitled-2022-10-11-2135](https://user-images.githubusercontent.com/124599/196148307-87a6cc6e-29e5-44ac-b54a-e800ec3d2a9a.png)

**Comment by suresh-kumar-ogilvy** [comment]
Thank you for the detailed info. 
I assume all the next.js "Site x" will send some kind of unique token i.e host  (since all custom domains will be hosted in one instance) to the API server and then the API server connects with drupal.

**Comment by shadcn** [comment]
Yeah. You can also use the request headers.

**Comment by suresh-kumar-ogilvy** [comment]
Thank you, since we are using the hostname as a unique key [[...slug]].tsx must use getServerSideProps to fetch the hostname?

is there any other way to fetch the hostname?

I did try this example https://github.com/vercel/examples/tree/main/edge-functions/hostname-rewrites some reason it doesn't support [[...slug].tsx

**Comment by shadcn** [comment]
@suresh-kumar-ogilvy What does your setup look like? Are you fetching data directly from Drupal?

**Comment by suresh-kumar-ogilvy** [comment]
yes, created a drupal custom module to get config by hostname
I did try using the API server but couldn't able to fetch the hostname from "Site A" if using getStaticPaths

**Comment by shadcn** [comment]
@sing78 Sure. Can you provide some more details about your architecture? Are you using Drupal in multi-site mode?

**Comment by sing78** [comment]
Hi,
I have to create multiple identical sites(same configuration, same theme, same content types but different content and users) with drupal and Next.js.
My first questions is:Do I need to create multiple databases or can I use a single database for all sites and if yes how?
If I use multiple databases, one drupal installation, and many Next.js front ends, how can I accomplish this?
Please provide as many details as possible and if possible useful links,
thanks

**Comment by theodorosploumis** [comment]
@sing78 What you need is:

- One codebase for the Backend (Drupal code, including configuration)
- One codebase for the Frontend (NextJS) but with different `.env` files for each frontend site
- Multiple databases for each site

In fact, this is a normal Drupal multisite.

The only important thing you have to decide is if every site is going to share the same **Drupal configuration** or not.

**Comment by sing78** [comment]
@theodorosploumis
Thanks for the help, that makes sense.
Cheers

**Comment by suresh-kumar-ogilvy** [comment]
@theodorosploumis 
Thanks a lot
This is just a clarification from my side for the below

- One codebase for the Frontend (NextJS) but with different .env files for each frontend site

- [ ] -> Each Frontend side will have its own node instance.

Thanks

**Comment by theodorosploumis** [comment]
@sing78 You can use https://www.npmjs.com/package/pm2 also which manages NodeJS apps.

**Comment by suresh-kumar-ogilvy** [comment]
@theodorosploumis  Thank you
